src/ui/public/state_management/state_storage/_tests__/state_hash.js

```diff
@@ -0,0 +1,29 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+
+import {
+  createStateHash,
+  HASH_PREFIX,
+  isStateHash,
+} from '../state_hash';
+
+describe('stateHash', () => {
+  describe('#createStateHash', () => {
+    describe('returns a string', () => {
+      it('beginning with the HASH_PREFIX', () => {
```

Comment:
It feels like these tests validate that the internal implementation a bit too much. Maybe focus a bit more on verifying the effect of the output/hash, rather than relying on the actual characters it contains. Consider these alternates:

 - `#createStateHash returns a hash` (tested using `#isStateHash`)
 - `#createStateHash returns the same hash for the same input`
 - `#createStateHash returns a different hash for every uniq input`
 - `#isStateHash returns true for hashes produced with #createHash`
 - `#isStateHash returns false for any value that rison can produce`
 - `#isStateHash returns false for any value that JSON.stringify can produce`

Those specs feel a bit more like they validate the behavior expected of this module and should continue to validate it even if the internal implementation changes dramatically.